src/sql/magic.py

```diff
@@ -410,18 +414,43 @@ def interactive_execute_wrapper(**kwargs):
             else:
                 with_ = self._store.infer_dependencies(command.sql_original, args.save)
                 if with_:
-                    command.set_sql_with(with_)
-                    display.message(
-                        f"Generating CTE with stored snippets: {pretty_print(with_)}"
-                    )
+                    query_type = self.get_query_type(command, original=True)
+
+                    if query_type != "SELECT":
+                        display.message_warning(
+                            f"Cannot use snippets with {query_type}"
+                        )
+                    else:
+                        command.set_sql_with(with_)
+                        display.message(
+                            f"Generating CTE with stored snippets: \
+{pretty_print(with_)}"
+                        )
                 else:
                     with_ = None
         else:
+            query_type = self.get_query_type(command, original=True)
             if args.with_:
                 raise exceptions.UsageError(
                     "Cannot use --with with CTEs, remove --with and re-run the cell"
                 )
 
+            if query_type != "SELECT":
+                display.message_warning("Cannot use snippets in CTEs")
+
+            dependencies = self._store.infer_dependencies(
+                command.sql_original, args.save
+            )
+
+            parentheses_content = re.findall(
+                r"\((.*)\)", command.sql_original.replace("\n", " ")
+            )
+            for query in parentheses_content:
+                for dependency in dependencies:
+                    if dependency in query:
+                        display.message_warning(
+                            f"Cannot use snippet {dependency} in CTE"
+                        )
```

Comment:
can you add some comments here? I'm not sure what this chunk of code is for

Reply:
I've added comments